Source/WebCore/Scripts/generate-process-sync-data.py

```diff
@@ -0,0 +1,302 @@
+        result.append('{')
+        result.append('    broadcastProcessSyncDataToOtherProcesses({ ProcessSyncDataType::%s, { data }});' % data.name)
+        result.append('}')
+
+    result.append('\n} // namespace WebCore\n')
+    return '\n'.join(result)
+
+
+_process_sync_data_header_prefix = """
+#pragma once
+
+namespace WebCore {
+"""
+
+_process_sync_data_header_suffix = """
+struct ProcessSyncData {
+    ProcessSyncDataType type;
+    ProcessSyncDataVariant value;
+};
+
+}; // namespace WebCore
+"""
+
+
+def sorted_qualified_types(synched_datas):
+    variant_type_set = set()
+    for data in synched_datas:
+        variant_type_set.add(data.fully_qualified_type)
+    return sorted(list(variant_type_set))
+
+
+def generate_process_sync_data_header(synched_datas):
+    result = []
+    result.append(_header_license)
+    result.append(_process_sync_data_header_prefix)
+
+    result.append("enum class ProcessSyncDataType : uint8_t {")
+    for data in sorted(synched_datas, key=lambda data: data.name):
+        result.append('    %s,' % data.name)
+    result.append("};")
+    result.append(" ")
+
+    result.append("using ProcessSyncDataVariant = std::variant<")
+
+    sorted_variant_types = sorted_qualified_types(synched_datas)
+    for variant_type in sorted_variant_types[:-1]:
+        result.append('    %s,' % variant_type)
+    result.append('    %s' % sorted_variant_types[-1])
+    result.append(">;")
+    result.append(_process_sync_data_header_suffix)
+    return '\n'.join(result)
+
+
+_serialization_in_license = """#
+# Copyright (C) 2024 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+"""
```

Comment:
Nit: I recommend you add "DO NOT EDIT; this file was generated by [scriptname]" to the generated output.